app/controllers/ops_controller/settings/label_tag_mapping.rb

```diff
@@ -0,0 +1,211 @@
+module OpsController::Settings::LabelTagMapping
+  extend ActiveSupport::Concern
+
+  def label_tag_mapping_edit
+    case params[:button]
+    when "cancel"
+      @lt_map = session[:edit][:lt_map] if session[:edit] && session[:edit][:lt_map]
+      if !@lt_map || @lt_map.id.blank?
+        add_flash(_("Add of new %{model} was cancelled by the user") %
+                    {:model => ui_lookup(:model => "ContainerLabelTagMapping")})
+      else
+        add_flash(_("Edit of %{model} \"%{name}\" was cancelled by the user") %
+                    {:model => ui_lookup(:model => "ContainerLabelTagMapping"), :name => @lt_map.label_name})
+      end
+      get_node_info(x_node)
+      @lt_map = @edit = session[:edit] = nil # clean out the saved info
+      replace_right_cell(@nodetype)
+    when "save", "add"
+      id = params[:id] ? params[:id] : "new"
+      return unless load_edit("label_tag_mapping_edit__#{id}", "replace_cell__explorer")
+      @lt_map = @edit[:lt_map] if @edit && @edit[:lt_map]
+      if @edit[:new][:label_name].blank?
+        add_flash(_("Label Name is required"), :error)
+      end
+      if @edit[:new][:category].blank?
+        add_flash(_("Category is required"), :error)
+      end
+      unless @flash_array.nil?
+        javascript_flash
+        return
+      end
+      if params[:button] == "add"
+        label_tag_mapping_add(@edit[:new][:entity], @edit[:new][:label_name], @edit[:new][:category])
+      else # save
+        label_tag_mapping_update(@lt_map.id, @edit[:new][:category])
+      end
+    when "reset", nil # Reset or first time in
```

Comment:
consider re-ordering the cases in first time; save/add; cancel order for easier reading.